src/pages/Lending/hooks/useGetEstimatedFeesQuery.ts

```diff
@@ -35,8 +36,8 @@ export const queryFn = async ({ queryKey }: { queryKey: EstimatedFeesQueryKey })
 export const useGetEstimatedFeesQuery = ({
   enabled,
   ...estimateFeesInput
-}: EstimateFeesInput & { enabled: boolean; disableRefetch?: boolean }) => {
-  const asset = useAppSelector(state => selectAssetById(state, estimateFeesInput.assetId))
+}: EstimateFeesInput & { enabled: boolean; disableRefetch?: boolean; feeAssetId: AssetId }) => {
```

Comment:
thinking about this more, we could reduce the need for `feeAssetId` and just select fee asset by asset id inside of here so that is abstracted away from the consumer. would reduce the need for currently declaring `assetId` and `feeAssetId` as it stands

Reply:
Could absolutely work, though we lose the declarativeness of `feeAssetId` introduced here - would be inclined to keep it for that reason

Reply:
yeah I could see both ways, one way declarative `feeAssetId` is nice because it states what it is, but also we now have `assetId` still which is not used anymore (I don't think) that is a bit confusing. And also need to add selectors in all consumers, whereas if just passing `assetId` and the hook handling it to get fee asset for fee things, means we don't have to think about it anymore

Reply:
`assetId` is still used by `estimateFees` which is the reason why it wasn't removed - it's passed in `estimateFeesArgs` to `estimateFees`:
https://github.com/shapeshift/web/blob/08ff42cd1fde2eb6ce3eb3cb4f21a692183ada3f/src/components/Modals/Send/utils.ts#L45

In this case, I believe *forcing* us to think about it is not a bad thing as we always seem to have a myriad of bugs related to this

Reply:
fair, I guess part of the issue is how commingled things are with various fee estimation logic. my assumption is that once we tackle the unified fees, a lot of this confusion/weirdness will go away.